l10n_it_dichiarazione_intento/models/account_invoice.py

```diff
@@ -25,13 +25,12 @@ def _set_fiscal_position(self):
                     )
                 if not all_dichiarazioni:
                     return
-                valid_date = fields.Date.from_string(invoice.date_invoice) \
-                    or fields.Date.context_today(invoice)
+                valid_date = invoice.date_invoice or fields.Date.context_today(invoice)
 
                 dichiarazioni_valide = all_dichiarazioni.filtered(
                     lambda d:
                     fields.Date.from_string(d.date_start)
-                    <= valid_date
+                    <= fields.Date.from_string(valid_date)
```

Comment:
In questo modo lo stesso parsing viene ripetuto per ogni dichiarazione, riesci a fare il parsing una volta sola invece?

Reply:
fatto, grazie